File: backend/tour-design-api/src/main/java/com/tourdesign/platform/service/PackageSpotService.java
package com.tourdesign.platform.service;

import com.tourdesign.platform.model.PackageSpotModel;
import java.util.List;
import java.util.Optional;

public interface PackageSpotService {

    List<PackageSpotModel> list();

    PackageSpotModel create(PackageSpotModel obj);

    Optional<PackageSpotModel> search(Long id);

    Optional<PackageSpotModel> update(Long id, PackageSpotModel obj);

    boolean delete(Long id);
}

File: frontend/tour-design-web/src/app/services/package-spot.service.ts
import { Injectable } from '@angular/core';
import { ApiService } from './api.service';
import { PackageSpotModel } from '../models/package-spot.model';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class PackageSpotService {
  private baseUrl = '/package-spot';

  constructor(private api: ApiService) {}

  list(): Observable<PackageSpotModel[]> {
    return this.api.get<PackageSpotModel[]>(this.baseUrl);
  }

  search(id: number): Observable<PackageSpotModel> {
    return this.api.get<PackageSpotModel>(`${this.baseUrl}/${id}`);
  }

  create(obj: PackageSpotModel): Observable<PackageSpotModel> {
    return this.api.post<PackageSpotModel>(this.baseUrl, obj);
  }

  update(id: number, obj: PackageSpotModel): Observable<PackageSpotModel> {
    return this.api.put<PackageSpotModel>(`${this.baseUrl}/${id}`, obj);
  }

  delete(id: number): Observable<void> {
    return this.api.delete<void>(`${this.baseUrl}/${id}`);
  }
}
